command/action.go

```diff
@@ -71,20 +80,31 @@ func runAction(args []string, provider, filepath, yamlInline string, subs map[st
 			} else {
 				value, err = prov.CallProvider(provider, spec.Path)
 				if err != nil {
-					fmt.Println(err.Error())
-					os.Exit(1)
+					results <- Result{key, err}
+					wg.Done()
+					return
 				}
 			}
 			envvar := formatForEnv(key, value, spec, &tempFactory)
-			results <- envvar
+			results <- Result{envvar, nil}
 			wg.Done()
 		}(key, spec)
 	}
 	wg.Wait()
 	close(results)
 
+	EnvLoop:
 	for envvar := range results {
-		env = append(env, envvar)
+		if envvar.error == nil {
+			env = append(env, envvar.string)
+		} else {
+			for i := range ignores {
+				if ignores[i] == envvar.string {
+					continue EnvLoop
+				}
+			}
+			return envvar.string, envvar.error
```

Comment:
I don't think we need to manually print at all, we probably should just hook io.Stderr to the command so it prints out as it executes